sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/SparkFileDataObject.scala

```diff
@@ -109,12 +109,16 @@ trait SparkFileDataObject extends HadoopFileDataObject
    * If a user-defined schema is returned, it overrides any schema inference. If no user-defined schema is set, the
    * schema may be inferred depending on the configuration and type of data frame reader.
    *
-   * @return The schema to use for the data frame reader when reading from the source.
+   * @return The consolidated schema to validate the schema on read and write, and for the data frame reader when reading from the source.
    */
   def getSchema(implicit context: ActionPipelineContext): Option[SparkSchema] = {
     _schemaHolder = _schemaHolder.orElse(
-        // get defined schema
-        schema.map(_.convert(typeOf[SparkSubFeed]).asInstanceOf[SparkSchema])
+        // get defined schema, add potentially missing partition columns as type string
+        schema.map { s =>
+          partitions.foldLeft(s.convert(typeOf[SparkSubFeed])) {
+              case (schema,col) => schema.addIfNotExists(col, SparkSimpleDataType(StringType))
```

Comment:
Should we find a way to get the actual datatype instead of String, or is it not possible at this point?

Reply:
Default for hadoop partition layout is string, other datatypes can cause strange problems.
-> It is best to leave it as string.